### PR TITLE
Swarm csi

### PR DESCRIPTION
**Pull request by Gradlon**
Build scripts and Documentation for docker Swarm CSI.

### CODE REVIEW

File: deploy/swarm/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+| `GID_MAP`             | Map local GID to GID on filer, comma-separated `<local_gid>:<filer_gid>`                               |
+
+## Mounts
+
+| Mount Destination   | Source Path     | Description                  |
+|---------------------|-----------------|------------------------------|
+| `/node_hostname`    | `/etc/hostname` | Used to get the nodename     |
+| `/tmp`              | `/tmp`          | Used for caching             |
+
+## Usag
```
Comment (chrislusf) [inline deploy/swarm/README.md]:
```suggestion
## Usage
```

File: cmd/seaweedfs-csi-driver/swarm/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| `GID_MAP`             | Map local GID to GID on filer, comma-separated `<local_gid>:<filer_gid>`                               |
+
+## Mounts
+
+| Mount Destination   | Source Path     | Description                  |
+|---------------------|-----------------|------------------------------|
+| `/node_hostname`    | `/etc/hostname` | Used to get the nodename     |
+| `/tmp`              | `/tmp`          | Used for caching             |
+
+## Usag
```
Comment (chrislusf) [inline cmd/seaweedfs-csi-driver/swarm/README.md]:
```suggestion
## Usage
```

File: cmd/seaweedfs-csi-driver/swarm/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+docker build --platform ${ARCH} --build-arg BASE_IMAGE=chrislusf/seaweedfs-csi-driver:${VERSION} --build-arg ARCH=$ARCH -t seawadd-csi_tmp_img .
+mkdir -p ./plugin/rootfs
+cp config.json ./plugin/
+docker container create --name seawadd-csi_tmp seawadd-csi_tmp_img 
+docker container export seawadd-csi_tmp | tar -x -C ./plugin/rootfs
+docker container rm -vf seawadd-csi_tmp 
+docker image rm seawadd-csi_tmp_img 
+
+docker plugin disable gradlon/swarm-csi-swaweedfs:v1.2.0
```
Comment (chrislusf) [inline cmd/seaweedfs-csi-driver/swarm/README.md]:
```suggestion
docker plugin disable ${PREFIX}/swarm-csi-swaweedfs:v1.2.0
```

File: deploy/swarm/README.md
Comment (chrislusf) [inline deploy/swarm/README.md:1]: this file mostly duplicates content in` cmd/seaweedfs-csi-driver/swarm/README.md`

maybe just remove this, or add a reference to the other file?

**Comment by chrislusf** [comment]
Thanks for this PR! I wonder whether there are some way to add an integration test for this, similar to https://github.com/seaweedfs/seaweedfs/blob/master/.github/workflows/e2e.yml , which can also serve as a usage example.

File: deploy/swarm/README.md
Comment (Gradlon) [reply deploy/swarm/README.md:1]: Sure.
Can you do this?

**Comment by chrislusf** [comment]
build failed https://github.com/seaweedfs/seaweedfs-csi-driver/actions/runs/10035448679/job/27731471376

**Comment by Gradlon** [comment]
I will delete my fork.

About e2e Test.
A real E2E test would mean spinning up a docker swarm cluster.
We would maybe need to discuss/you tel me you would go about this.

Thank you for this Project, I am using seaweed in my Cluster, and I am pretty happy with it!

If I can contribute more, I will.

One thing that I see is that the log output of the plugin appears as errors in docker logs.
This is not a big issue but also not right ;)

can this be adjusted?

**Comment by chrislusf** [comment]
> A real E2E test would mean spinning up a docker swarm cluster. We would maybe need to discuss/you tel me you would go about this.

Yes. That would be nice. I do not know much here though.

> One thing that I see is that the log output of the plugin appears as errors in docker logs. can this be adjusted?

Yes. `weed -logtostderr=false ...`

**Comment by blib** [comment]
please note that in docker 27.x.x CSI plugin interface is broken

https://github.com/moby/moby/issues/48133
https://github.com/moby/swarmkit/pull/3181

**Comment by Gradlon** [comment]
@blib You are right.
This is the issue for several ported CSI Plugins, e.g. Hetzner CSI is also broken.
@chrislusf Is there a fix planet?

**Comment by blib** [comment]
it's not a seaweedfs problem. I submitted PR https://github.com/moby/swarmkit/pull/3181 you can up vote it :)

**Comment by Gradlon** [comment]
@blib Ah thank you for the correction, i got that wrong then.
I upvoted the PR ;)

**Pull request merged**